<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>881</width>
    <height>764</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>2</number>
    </property>
    <property name="leftMargin">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="rightMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item>
     <widget class="QFrame" name="frameToolbar">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>4</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QToolButton" name="buttonCalc">
         <property name="toolTip">
          <string>Recalculate current position</string>
         </property>
         <property name="text">
          <string>Calc</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="buttonTop">
         <property name="toolTip">
          <string>Go to top-level position</string>
         </property>
         <property name="text">
          <string>Top</string>
         </property>
         <property name="icon">
          <iconset theme="QIcon::ThemeIcon::GoHome"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="buttonPrev">
         <property name="toolTip">
          <string>Go to previous position</string>
         </property>
         <property name="text">
          <string>Prev</string>
         </property>
         <property name="icon">
          <iconset theme="QIcon::ThemeIcon::GoPrevious"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="buttonNext">
         <property name="toolTip">
          <string>Go to next position</string>
         </property>
         <property name="text">
          <string>Next</string>
         </property>
         <property name="icon">
          <iconset theme="QIcon::ThemeIcon::GoNext"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="buttonZoomIn">
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset theme="QIcon::ThemeIcon::ZoomIn"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="buttonZoomOut">
         <property name="toolTip">
          <string>Zoom out by factor of 2</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset theme="QIcon::ThemeIcon::ZoomOut"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>RES:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEditResolution">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="baseSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Primary axis resolution (pixels)</string>
         </property>
         <property name="text">
          <string>1000</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>ITER:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEditIterMax">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Maximum iterations</string>
         </property>
         <property name="text">
          <string>500</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>TC:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBoxThreadCount">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777214</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Compute thread count</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>24</number>
         </property>
         <property name="value">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>SS:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBoxSuperSample">
         <property name="toolTip">
          <string>Super sampling factor</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>AR:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxAspect">
         <property name="toolTip">
          <string>Select zoom aspect ratio</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>PAL:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxPalette">
         <property name="toolTip">
          <string>Select color palette</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="buttonPaletteEdit">
         <property name="toolTip">
          <string>Edit current palette</string>
         </property>
         <property name="text">
          <string>Edit</string>
         </property>
         <property name="icon">
          <iconset theme="QIcon::ThemeIcon::DocumentProperties"/>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QToolButton" name="buttonLoad">
         <property name="toolTip">
          <string>Load image from file</string>
         </property>
         <property name="text">
          <string>Load</string>
         </property>
         <property name="icon">
          <iconset theme="QIcon::ThemeIcon::DocumentOpen"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="buttonSave">
         <property name="toolTip">
          <string>Save current image to file</string>
         </property>
         <property name="text">
          <string>Save</string>
         </property>
         <property name="icon">
          <iconset theme="QIcon::ThemeIcon::DocumentSave"/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame_2">
      <property name="frameShape">
       <enum>QFrame::Shape::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Scale:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="sliderPalScale">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>20</number>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_5">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Offset:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="sliderPalOffset">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frameViewer">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>881</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionExportImage"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menu_File"/>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="toolTip">
    <string>Exit Mandelbrot application</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionExportImage">
   <property name="text">
    <string>Export &amp;Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonCalc</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>calculate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBoxPalette</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>paletteSelect(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>476</x>
     <y>34</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderPalOffset</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>paletteOffsetSliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderPalScale</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>paletteScaleSliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonSave</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>613</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonTop</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomTop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>66</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonPrev</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>prev()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonNext</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>next()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>147</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBoxAspect</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>aspectChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>589</x>
     <y>34</y>
    </hint>
    <hint type="destinationlabel">
     <x>439</x>
     <y>381</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonPaletteEdit</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>paletteEdit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>720</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>439</x>
     <y>381</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonLoad</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>838</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>381</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonZoomOut</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>381</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonZoomIn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomIn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>381</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>calculate()</slot>
  <slot>paletteOffsetSliderChanged(int)</slot>
  <slot>paletteScaleSliderChanged(int)</slot>
  <slot>saveImage()</slot>
  <slot>zoomTop()</slot>
  <slot>prev()</slot>
  <slot>next()</slot>
  <slot>aspectChange(int)</slot>
  <slot>paletteSelect(QString)</slot>
  <slot>paletteEdit()</slot>
  <slot>loadImage()</slot>
  <slot>zoomOut()</slot>
  <slot>zoomIn()</slot>
 </slots>
</ui>
